### PR TITLE
BHoM_Engine: Additional check for assembly qualified type without version for Create EngineType

### LINKED ISSUE #3398: BHoM_Engine: Fix issue with finding correct engine type for duplicate namespace across repositories
#### Description:

Problem introduced in https://github.com/BHoM/BHoM_Engine/pull/3389 for edge case of: 
Deserialising an engine type that has a assembly qualified name, but was serialised in a previous major milestone, and has another duplicate in terms of namespace and typename in another assembly.

#### Expected behaviour:

For method Create EngineType to atempt to filter based on AssemblyName, without version specified, in case mroe than 2 items have been found for a particular type name.

### PR DESCRIPTION
**Pull request by IsakNaslundBh**
### Issues addressed by this PR

Closes #3398 

Add additional check for assembly qualified name without version. Important for edgecase of same namespace and typename, from different assemblies for types serialized in an older major version. Only runs this additional step if more than one type is found.

Generally having this duplication should most likely be avoided, but as that has not been enforced, and was not an issue before #3389 which fixed other critical issues, but introduced this edge case problem, we should still attempt to make sure it works after that fix.

### Test files

All query components should deserialise correctly:

[On Sharepoint](https://burohappold.sharepoint.com/:u:/r/sites/BHoM/02_Current/12_Scripts/03_Alpha/BuroHappoldEngineering/ModelLaundry_Toolkit/BuroHappold_BHoM_v3.3.beta_Structures_ML_BasicWorkFlow.gh?csf=1&web=1&e=6E7xl3)https://burohappold.sharepoint.com/:u:/r/sites/BHoM/02_Current/12_Scripts/03_Alpha/BuroHappoldEngineering/ModelLaundry_Toolkit/[BuroHappold_BHoM_v3.3.beta_Structures_ML_BasicWorkFlow.gh](https://burohappold.sharepoint.com/:u:/r/sites/BHoM/02_Current/12_Scripts/03_Alpha/BuroHappoldEngineering/ModelLaundry_Toolkit/BuroHappold_BHoM_v3.3.beta_Structures_ML_BasicWorkFlow.gh?csf=1&web=1&e=6E7xl3)?csf=1&web=1&e=6E7xl3

### CODE REVIEW

**Comment by IsakNaslundBh** [comment]
@BHoMBot check compliance

**Comment by IsakNaslundBh** [comment]
@BHoMBot check unit-tests
@BHoMBot check required

File: BHoM_Engine/Create/Type/EngineType.cs [hunk 1]
```diff
@@ -49,14 +49,34 @@ public static Type EngineType(string name, bool silent = false, bool takeFirstIf
...
+                    unQualifiedName = name;
+                    assemblyName = null;
+                }
+
                 types = Global.EngineTypeList.Where(x => x.FullName == unQualifiedName).ToList();
+
+                //If more than one type found, and assembly name was found, try use it to further filter
+                if (types.Count > 1 && assemblyName != null)
+                {
+                    List<Type> assemblyMatches = types.Where(x => x.Assembly.FullName.Split(',')[0] == assemblyName).ToList();  //Assembly fullname contains both name and version. Split removes the verison
```
Comment (pawelbaran) [inline BHoM_Engine/Create/Type/EngineType.cs:76]: You could use the native members of `Assembly` class to get the name
```suggestion
                    List<Type> assemblyMatches = types.Where(x => x.Assembly.GetName().Name == assemblyName).ToList();  //Assembly fullname contains both name and version. Split removes the verison
```

**Review by pawelbaran — COMMENTED** [review]
The PR looks good, I tested it both in GH and Revit with positive results. One minor code comment left, besides that all good 👍

File: BHoM_Engine/Create/Type/EngineType.cs (hunk 1 above)
Comment (IsakNaslundBh) [reply BHoM_Engine/Create/Type/EngineType.cs]: Ahh, great. Only looked for the Name on the assembly - missed the GetName(). Thanks. I commited separately to get rid of the comment that does not make sense any longer, rather than commiting your suggestion.

**Comment by IsakNaslundBh** [comment]
@BHoMBot check compliance
@BHoMBot check required
@BHoMBot check unit-tests

**Comment by pawelbaran** [comment]
@BHoMBot check ready-to-merge
